src/plaid/plaid.service.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable } from '@nestjs/common'
+import { ModelId } from '../core/model'
+import { EncryptionService } from '../encryption/encryption.service'
+import { PlaidItem, tableName } from './plaidItem.entity'
+import { InjectRepository } from '@nestjs/typeorm'
+import { Repository } from 'typeorm'
+import plaid, { Account, Client } from 'plaid'
+import { ConfigService } from '@nestjs/config'
+import { IPlaidConfig } from '../config/plaidConfig'
+
+@Injectable()
+export class PlaidService {
+  private readonly client: Client
+  constructor(
+    private readonly encryptionService: EncryptionService,
+    private readonly configService: ConfigService,
+    @InjectRepository(PlaidItem) private readonly plaidSessionRepo: Repository<PlaidItem>,
+  ) {
+    const config = this.configService.get<IPlaidConfig>('plaid')
+    if (!config) {
+      throw new Error('Missing Plaid Config')
+    }
+    this.client = new plaid.Client(
+      config.clientId,
+      config.secret,
+      config.publicKey,
+      plaid.environments[config.env],
+      { version: '2019-05-29', clientApp: 'Bitmatica' },
+    )
+  }
+
+  async exchangePublicToken(userId: ModelId, publicToken: string): Promise<string> {
+    const response = await this.client.exchangePublicToken(publicToken)
+    const item = new PlaidItem()
+    item.userId = userId
+    item.itemId = response.item_id
+    item.accessToken = await this.encryptionService.encrypt(
+      response.access_token,
+      PlaidService.encryptionContext(userId, response.item_id),
+    )
+    await this.plaidSessionRepo.save(item)
+    return item.itemId
+  }
+
+  async getPlaidAccounts(userId: ModelId, itemId: string): Promise<Array<Account>> {
+    const accessToken = await this.getAccessToken(userId, itemId)
+    if (!accessToken) {
+      return Promise.reject('Access token invalid')
+    }
+
+    return (await this.client.getAccounts(accessToken)).accounts
+  }
+
+  async getAccessToken(userId: ModelId, itemId: string): Promise<string | undefined> {
+    // TODO: cache accessToken in-memory to avoid decryption calls?
+    const item = await this.plaidSessionRepo.findOne({ userId, itemId })
+    if (!item) {
+      return undefined
+    }
+    return await this.encryptionService.decrypt(
```

Comment:
This is totally fine as is, but just some comments for js slickness:
you could remove lines 57-59 and do `return item && this.encryptionService.decrypt(...`
if item is undefined, it will return `undefined`, although, that might look a little awkward with the await. 
I also prefer `if (!item) return` as a one-liner - a naked return returns `undefined`, so no need to include it explicitly

Reply:
Thanks!  Updated to the first bullet.